pages/gallery/gallery.js

```diff
@@ -0,0 +1,420 @@
+        versions: settings.versions.java,
+        editions: settings.editions.map(e => e.toLowerCase())
+      },
+      current: {
+        resolution: settings.resolutions[0],
+        tag: 'All',
+        version: settings.versions.java[0],
+        edition: settings.editions[0],
+        search: null
+      },
+      dataJSON: {},
+      displayed: {
+        paths: {},
+        uses: {},
+        textures: {},
+        contributions: {
+          c32: [],
+          c64: []
+        },
+        contributors: {}
+      },
+      displayedResults: 20,
+      displayedTextures: [],
+      modalTextureID: null,
+      modalTextureObj: {},
+      modalOpen: false,
+    }
+  },
+  computed: {
+    resolutions() { return this.options.resolutions },
+    tags() { return ['all', ...this.options.tags] },
+    versions() { return this.options.versions },
+    editions() { return this.options.editions },
+
+    resolution() { return this.current.resolution },
+    tag() { return this.current.tag },
+    version() { return this.current.version },
+    edition() { return this.current.edition },
+    search() { return this.current.search },
+  },
+  watch: {
+    '$route.params': {
+      handler(params) {
+        this.current.resolution = params.resolution
+        this.current.edition = params.edition
+        this.current.version = params.version
+        this.current.tag = params.tag
+        this.current.search = params.search
+
+        this.updateJSON(true)
+      },
+      deep: true,
+      immediate: true
+    }
+  },
+  created() {
+    // directly add the search to the search bar if there is a search parameter inside the router
+    this.current.search = this.$route.params.search ? this.$route.params.search : undefined
+    this.options.versions = this.$route.params.edition ? settings.versions[this.$route.params.edition] : settings.versions[0]
+
+    window.addEventListener('scroll', this.handleScroll)
+  },
+  methods: {
+    openModal(id) {
+      this.modalTextureID = id
+      this.modalOpen = true
+
+      axios.get('/gallery/dialog/' + id)
+        .then(res => {
+          this.modalTextureObj = res.data
+        })
+    },
+    closeModal() {
+      this.modalOpen = false
+      this.modalTextureID = null
+      this.modalTextureObj = {}
+    },
+    splittedTextures() {
+      const result = []
+      const length = Object.keys(this.displayed.textures).length
+
+      for (let i = 0; i < length; i++) {
+        result.push(Object.values(this.displayed.textures)[i])
+      }
+
+      return result
+    },
+    getAuthor(textureID) {
+      let contributionsHTML = ''
+
+      let contributions = this.displayed.contributions[this.current.resolution === '32x' ? 'c32' : 'c64'][textureID]
+      if (contributions) {
+        const timestamp = contributions.reduce((a, b) => a = a > b.date ? a : b.date, 0)
+        const contribution = contributions.filter(el => el.date == timestamp)[0]
+
+        contributionsHTML = `
```

Comment:
make a component out of this, please no inline html

Reply:
I insist, if you have to do this, it means that you don't fully understand js framework power

Reply:
I can't because of this line
`v-tooltip.right-start="{content: () => getAuthor(texture.textureID), html: true}"` which cannot take vue component